src/os/FileStore.cc

```diff
@@ -4180,7 +4196,8 @@ int FileStore::_setattrs(coll_t cid, const ghobject_t& oid, map<string,bufferptr
 	continue;
     }
     omap_remove.insert(p->first);
-    inline_set.insert(*p);
+    if (!inline_set.count(p->first))
+      inline_set.insert(make_pair(p->first, make_pair(p->second, -1)));
```

Comment:
@liewegas read all xattrs then write xattrs is really a awful impl now. Do we consider to make internal xattrs("_", "snapset") not go through this? I think caller could give a hint.